### PR TITLE
Add a generator for an Extension pack

### PR DESCRIPTION
**Pull request by egamma**
Addresses https://github.com/Microsoft/vscode-generator-code/issues/63.

### CODE REVIEW

**Comment by aeschli** [comment]
- when running against vscode without any extension installed, I get the following package.json
    "extensionDependencies": [ 
        ""
    ]
}
- disabled extensions are also listed
- When I don't have 'code' installed I get
Error: Uncaught, unspecified "error" event. (undefined)
    at emit (events.js:157:17)
    at Immediate._onImmediate (/home/aeschli/workspaces/vscode-generator-code/node_modules/yeoman-generator/lib/base.js:446:16)
    at tryOnImmediate (timers.js:534:15)
    at processImmediate [as _immediateCallback] (timers.js:514:5)

I suggest to print an error to the console instead and continue.
- For completeness, there should also be a test case (at least for the 'no' (don't read existing)) case. See https://github.com/Microsoft/vscode-generator-code/blob/master/test/test.js.

**Comment by egamma** [comment]
> when running against vscode without any extension installed, I get the following package.json

Fixed

> disabled extensions are also listed

Currently there is no such command line option. Filed https://github.com/Microsoft/vscode/issues/19083

> When I don't have 'code' installed I get

Fixed.

> For completeness, there should also be a test case (at least for the 'no' (don't read existing)) case. 

Added test.

Also added a warning at the end that user reviews the `extensionDependencies` attribute.

**Comment by egamma** [comment]
@aeschli pls review again, the generator should now be in good shape.

File: generators/app/index.js [hunk 1]
```diff
@@ -304,6 +310,41 @@ module.exports = yeoman.Base.extend({
...
+            }).then(function (addExtensionsAnswer) {
+
+                generator.extensionConfig.extensionList = ["publisher.extensionName"];
+
+                if (addExtensionsAnswer.addExtensions) {
+                    return new Promise(function (resolve, reject) {
+                        childProcess.exec('code --list-extensions', function(error, stdout, stderr) {
+                            if (error) {
+                                generator.env.error("Problems starting Code: " + error);
+                                return;
```
Comment (aeschli) [inline generators/app/index.js]: 'resolve' should be called before the return

File: test/test.js [hunk 2]
```diff
@@ -280,4 +280,42 @@ describe('test theme generator', function () {
...
+
+    it('extension-pack', function (done) {
+        helpers.run(path.join(__dirname, '../generators/app'))
+            .withPrompts({
+                addExtensions: 'n',
+                type: 'ext-extensionpack',
+                name: 'testExtensionPack',
+                displayName: 'Test Extension Pack',
+                description: 'My Test Extension Pack',
+                publisher: 'Microsoft'
```
Comment (aeschli) [inline test/test.js]: Add 'addExtensions' here with true or false

Comment (aeschli) [inline test/test.js]: Both variants (true and false) are interesting as test case.

**Comment by aeschli** [comment]
PR merged! Thanks @egamma!

**Comment by egamma** [comment]
Thanks @aeschli for the careful review and guidance.

**Pull request merged**